jujugui/static/gui/src/app/components/entity-details/content/diagram/diagram.js

```diff
@@ -19,29 +19,67 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 'use strict';
 
 class EntityContentDiagram extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      url: this.props.getDiagramURL(this.props.id)
+    };
+  }
+
+  _handleClose() {
+    this.props.clearLightbox();
```

Comment:
You can't remove a component from within itself, so I followed the convention used for the modals etc.